realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -166,6 +167,7 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
      * @throws UnsupportedOperationException if the required RxJava framework is not on the classpath.
      * @see <a href="https://realm.io/docs/java/latest/#rxjava">RxJava and Realm</a>
      */
+    @OptionalAPI(dependencies = {"rx.Observable"})
```

Comment:
@nhachicha This requires `compiles provided 'io.reactivex:rxjava:1.1.0'` to transformer's `build.gradle` since it needs the class's information when running the transformer. That will also couples the transformer's with the dependencies -- whenever a new optional dependency added, it has to be compiled into transformer.